spec/concerns/redirect_behavior_spec.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+class Controller < AnonymousController
+  include RedirectBehavior
+end
+
+RSpec.describe RedirectBehavior do
+  let(:controller) { Controller.new }
+  let(:exercise) { create(:hello_world) }
+  let(:contributor) { create(:external_user) }
+  let(:submission) { create(:submission, exercise:, contributor:, score: 1) }
+
+  before do
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(contributor)
+    allow(Submission).to receive(:find).and_return(submission)
+  end
+
+  describe '#redirect_after_submit' do
+    context 'when community solution is submitted' do
+      before do
+        allow(described_class).to receive(redirect_to_community_solution?).and_return(true)
+      end
+
+      it 'redirects to community solution' do
+        expect_redirect { edit_community_solution_path(community_solution, lock_id: community_solution_lock.id) }
+        # expect(controller).to receive(:redirect_to).with(edit_community_solution_path(community_solution, lock_id: community_solution_lock.id))
+      end
+    end
```

Comment:
You will probably need to move all tests to `specs/submissions_controller_spec.rb`. Then, this test would read as:

```suggestion
    context 'when a community solution is offered' do
      let!(:community_solution) { CommunitySolution.create(exercise:) }

      before do
        allow(Java21Study).to receive(:allow_redirect_to_community_solution?).and_return(true)
        post :finalize, params: {id: submission.id}
      end

      expect_redirect { edit_community_solution_path(community_solution, lock_id: CommunitySolutionLock.last) }
    end
```